controllers/numaresourcesoperator_controller.go

```diff
@@ -207,25 +208,27 @@ func (r *NUMAResourcesOperatorReconciler) degradeStatus(ctx context.Context, ins
 	return ctrl.Result{}, nil
 }
 
-func (r *NUMAResourcesOperatorReconciler) reconcileResourceAPI(ctx context.Context, instance *nropv1.NUMAResourcesOperator, trees []nodegroupv1.Tree) (bool, ctrl.Result, string, error) {
+func (r *NUMAResourcesOperatorReconciler) reconcileResourceAPI(ctx context.Context, instance *nropv1.NUMAResourcesOperator, trees []nodegroupv1.Tree) (bool, ctrl.Result, conditionInfo, error) {
```

Comment:
I don't see a reason to keep the returned error, it is already part of the returned `conditionInfo`

Reply:
same below in the other reconciliation substeps

Reply:
that's a good point. What it should go is not the top level error though, which is used to report to the upper layers, but the inner error in `ConditionInfo`. Let's see what I can do.